velox/exec/Spill.h

```diff
@@ -0,0 +1,350 @@
+  // different sorted run and must be ordered between themselves but not
+  // with respect to calls before flush(). If 'close' is true, write must
+  // not be called and the file set can only be read.
+  void flush(bool close);
+
+ private:
+  const RowTypePtr type_;
+  const std::string path_;
+  const uint64_t targetBatchSize_;
+  const uint64_t targetFileSize_;
+  memory::MemoryPool& pool_;
+  memory::MappedMemory& mappedMemory_;
+  std::unique_ptr<VectorStreamGroup> batch_;
+  bool isOpen_ = false;
+  std::vector<std::unique_ptr<SpillFile>> files_;
+};
+
+// Represents all spilled data of an operator, e.g. order by or group
+// by. This has one SpillFileList per partition of spill data.
+class SpillState {
+ public:
+  // Constructs a SpillState. 'type' is the content RowType. 'path' is the file
+  // system path prefix. 'bits' is the hash bit field for partitioning data
+  // between files. This also gives the maximum number of partitions.
+  // 'targetFileSize' is the target size of a single file. 'targetBatchSize is
+  // the target number of rows in a single RowVector written to a spill file.
+  // 'pool' and 'mappedMemory' own the memory for state and results.
+  SpillState(
+      RowTypePtr type,
+      const std::string& path,
+      HashBitRange bits,
+      uint64_t targetFileSize,
+      uint64_t targetBatchSize,
+      memory::MemoryPool& pool,
+      memory::MappedMemory& mappedMemory)
+      : type_(type),
+        path_(path),
+        bits_(bits),
+        fieldMask_(((1UL << (bits_.end - bits_.begin))) - 1),
+        targetFileSize_(targetFileSize),
+        targetBatchSize_(targetBatchSize),
+        pool_(pool),
+        mappedMemory_(mappedMemory) {}
+
+  // Returns which spill partition 'hash' falls. Returns -1 if the
+  // partition of 'hash' is has not been started.
+  int32_t partition(uint64_t hash) {
+    auto field = (hash >> bits_.begin) & fieldMask_;
+    return field < numPartitions_ ? field : -1;
+  }
+
+  int32_t numPartitions() const {
+    return numPartitions_;
+  }
+
+  // Sets how many of the spill partitions are in use.
+  void setNumPartitions(int32_t numPartitions);
+
+  // Returns how many ways spilled data can be partitioned.
+  uint16_t maxPartitions() {
+    return 1 << (bits_.end - bits_.begin);
+  }
+
+  uint64_t targetFileSize() const {
+    return targetFileSize_;
+  }
+
+  uint64_t targetBatchSize() const {
+    return targetBatchSize_;
+  }
+
+  memory::MemoryPool& pool() {
+    return pool_;
+  }
+
+  // Appends data to 'partition'. The rows  given by 'indices'  must be sorted
+  // and must hash to 'partition'.
+  void appendToPartition(
+      uint16_t partition,
+      const RowVectorPtr& rows,
+      const folly::Range<IndexRange*> indices);
+
+  // Finishes a sorted run for 'partition'. If write is called for 'way'
+  // again, the data does not have to be sorted relative to the data
+  // written so far.
+  void finishWrite(uint16_t partition) {
+    files_[partition]->flush(true);
```

Comment:
Why do you pass "true" to flush? The comment suggests that the caller may continue writing to this partition.